current/advanced/block-import.md

```diff
@@ -0,0 +1,90 @@
+import steps sequentially. It does, however, abstract the notion of verification through the use of
+the
+[`Verifier`](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sp_consensus/import_queue/trait.Verifier.html)
+trait.
+
+Any consensus engine that relies on the `BasicQueue` must implement the `Verifier` trait. The
+`Verifier` is typically responsible for tasks such as checking
+[inherent data](https://www.substrate.io/kb/learn-substrate/extrinsics#inherents), and ensuring that
+the block is signed by the appropriate authority.
+
+## The Block Import Trait
+
+When the import queue is ready to import a block, it passes the block in question to a method
+provided by the
+[`BlockImport` trait](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sp_consensus/block_import/trait.BlockImport.html).
+This `BlockImport` trait provides the behavior of importing a block into the node's local state
+database.
+
+One implementor of the `BlockImport` trait that is used in every Substrate node is the
+[`Client`](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sc_service/client/index.html) which
+contains the node's entire block database. When a block is imported into the client it is added to
+the main database of blocks that the node knows about.
+
+## The Block Import Pipeline
+
+In the simplest cases, such as
+[manual-seal](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sc_consensus_manual_seal/index.html)
+nodes, blocks are imported directly into the client. But most consensus engines will need to perform
+additional verification on incoming blocks, update their own local state databases, or both. To
+allow consensus engines this opportunity, it is common to wrap the client in another struct which is
+also `BlockImport`. This nesting leads to the term "block import pipeline".
+
+An example of this wrapping is the
+[`PowBlockImport`](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sc_consensus_pow/struct.PowBlockImport.html)
+which holds a reference to another type that is also `BlockImport`. This allows the PoW consensus
+engine to do it's own import-related bookkeeping and then pass the block to the nested
+`BlockImport`, probably the client. This pattern is also demonstrated in the
+[`AuraBlockImport`](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sc_consensus_aura/struct.AuraBlockImport.html),
+[`BabeBlockImport`](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sc_consensus_babe/struct.BabeBlockImport.html),
+and `GrandpaBlockImport`.
+
+`BlockImport` nesting need not be limited to one level. In fact, it is common for nodes that use
+both an authoring engine and a finality gadget to layer the nesting even more deeply. For example
+Polkadot's block import pipeline consists of a `GrandpaBlockImport` which wraps a `BabeBlockImport`
+which, itself, wraps the `Client`.
+
+## Learn More
+
+Several of the Recipes' nodes demonstrate the block import pipeline:
+
+- [Manual Seal](https://substrate.dev/recipes/3-entrees/manual-seal.html) - all blocks are valid so
```

Comment:
Looks like you give a little more explanation about Manual Seal here...may be nice to bring this up to the line where I had my other comment re Manual Seal.

Reply:
I removed that previous mention, which linked to the actual `ManualSealBlockImport` struct. This one links to the recipe for the manual seal node. But yes, it is explained in that recipe why manual seal doesn't need to have its own `BlockImport`. (Because it considers all blocks valid)